src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -93,6 +143,11 @@ class InternalFrame:
     # The length of data_column_index_names equals to number of multiindex levels.
     # For a 3-level MultiIndex, the value can be like ['A', 'B', 'C']
     data_column_index_names: tuple[LabelTuple, ...]
+    # Map from snowflake identifier to cached Snowpark pandas data type.
+    # The type is None if we don't know the Snowpark data type.
+    # n.b. that we map to SnowparkPandasType rather than to DataType, because
+    # we don't want to try tracking regular Snowpark Python types at all.
+    snowflake_quoted_identifier_to_type: dict[str, Optional[SnowparkPandasType]]
```

Comment:
InternalFrame is an Immutable class. We should not use mutable dict here, that will break the guarantee that we have right now.

Reply:
https://stackoverflow.com/questions/11014262/how-to-create-an-immutable-dictionary-in-python

Reply:
Switched to immutable `MappingProxyType `.